pynet/datasets/euaims.py

```diff
@@ -0,0 +1,840 @@
+        "lte": lambda x, y: x <= y,
+        "eq": lambda x, y: x == y,
+    }
+    for name, controls in qc.items():
+        for relation, value in controls.items():
+            if relation not in relation_mapper.keys():
+                raise ValueError("The relationship {} provided is not a \
+                    valid one".format(relation))
+            else:
+                new_idx = relation_mapper[relation](
+                    data["{}{}".format(prefix, name)], value)
+                idx_to_keep = idx_to_keep & new_idx
+    return data[idx_to_keep]
+
+
+def fetch_clinical(
+    datasetdir=SAVING_FOLDER, test_size=0.2, seed=42,
+    return_data=False, z_score=True,
+    drop_cols=["t1_site", "t1_ageyrs", "t1_sex", "t1:fsiq", "t1_group",
+               "t1_diagnosis", "mri", "t1_group_name"],
+    qc={"t1:fsiq": {"gte": 70}, "mri": {"eq": 1}, "qc": {"eq": "include"}},
+):
+    """ Fetches and preprocesses clinical data
+
+    Parameters
+    ----------
+    datasetdir: string
+        path to the folder in which to save the data
+    test_size: float, default 0.2
+        proportion of the dataset to keep for testing. Preprocessing models
+        will only be fitted on the training part and applied to the test
+        set. You can specify not to use a testing set by setting it to 0
+    seed: int, default 42
+        random seed to split the data into train / test
+    return_data: bool, default False
+        If false, saves the data in the specified folder, and return the
+        path. Otherwise, returns the preprocessed data and the
+        corresponding subjects
+    z_score: bool, default True
+        wether or not to transform the data into z_scores, meaning
+        standardizing and scaling it
+    drop_cols: list of string, see default
+        names of the columns to drop before saving the data
+    qc: dict, see default
+        keys are the name of the features the control on, values are the
+        requirements on their values (see the function apply_qc)
+
+    Returns
+    -------
+    path: string,
+        path to the training data, if return_data is False
+    path_test: string,
+        path to the testing data, if return_data is False and test_size > 0
+    X_train: numpy array,
+        Training data, if return_data is True
+    X_test: numpy array,
+        Test data, if return_data is True and test_size > 0
+    subj_train: numpy array,
+        Training subjects, if return_data is True
+    subj_test: numpy array,
+        Test subjects, if return_data is True and test_size > 0
+    """
+
+    data = pd.read_table(FILES["clinical"])
+
+    data_train = apply_qc(data, "", qc).sort_values("subjects")
+
+    X_train = data_train.drop(columns=drop_cols)
+
+    # Splits in train and test and removes nans
+    if test_size > 0:
+        X_train, X_test = train_test_split(
+            X_train, test_size=test_size, random_state=seed)
+
+        na_idx_test = (X_test.isna().sum(1) == 0)
```

Comment:
Why not using pd.dropna?

Reply:
Great question. I'm changing it ASAP.

Reply:
Actually I need these indices to retrieve the corresponding subject in the other dataframe. Is it ok to leave it this way?

Reply:
It's fine.